File: app/src/main/java/com/github/sewerina/meter_readings/DateConverter.java
package com.github.sewerina.meter_readings;

import androidx.room.TypeConverter;

import java.util.Date;

public class DateConverter {
    @TypeConverter
    public long fromDate(Date date) {
        return date.getTime();
    }

    @TypeConverter
    public Date fromTime(long time) {
        return new Date(time);
    }

}

File: app/src/main/java/com/github/sewerina/meter_readings/database/DateConverter.kt
package com.github.sewerina.meter_readings.database

import androidx.room.TypeConverter
import java.util.*

class DateConverter {
    @TypeConverter
    fun fromDate(date: Date): Long {
        return date.time
    }

    @TypeConverter
    fun fromTime(time: Long): Date {
        return Date(time)
    }
}